### ISSUE TITLE
[staging] firefox fails to build (seemingly after rust-1.56 update)

### ISSUE THREAD

**Issue by trofi** [NixOS/nixpkgs#144670]
On `staging` branch `firefox` fails to build as:

```
$ nix build -f. firefox
...
  cargo:warning=In file included from glsl-optimizer/src/compiler/glsl_types.cpp:25:
  cargo:warning=In file included from glsl-optimizer/src/mesa/main/macros.h:35:
  cargo:warning=In file included from glsl-optimizer/src/util/u_math.h:42:
  cargo:warning=In file included from glsl-optimizer/include/c99_math.h:37:
  cargo:warning=In file included from /build/firefox-94.0/obj-x86_64-pc-linux-gnu/dist/system_wrappers/math.h:3:
  cargo:warning=In file included from /nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/math.h:300:
  cargo:warning=In file included from /nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:36:
  cargo:warning=In file included from /build/firefox-94.0/obj-x86_64-pc-linux-gnu/dist/stl_wrappers/cmath:64:
  cargo:warning=In file included from /build/firefox-94.0/obj-x86_64-pc-linux-gnu/dist/system_wrappers/cmath:3:
  cargo:warning=/nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/cmath:532:46: error: declaration conflicts with target of using declaration already in scope
  cargo:warning=inline _LIBCPP_INLINE_VISIBILITY float       hypot(       float x,       float y,       float z ) { return sqrt(x*x + y*y + z*z); }
  cargo:warning=                                             ^
  cargo:warning=/nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/cmath:423:1: note: target of using declaration
  cargo:warning=using ::hypot _LIBCPP_USING_IF_EXISTS;
  cargo:warning=^
  cargo:warning=/nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/cmath:423:9: note: using declaration
  cargo:warning=using ::hypot _LIBCPP_USING_IF_EXISTS;
  cargo:warning=        ^
  cargo:warning=/nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/cmath:532:108: error: reference to unresolved using declaration
  cargo:warning=inline _LIBCPP_INLINE_VISIBILITY float       hypot(       float x,       float y,       float z ) { return sqrt(x*x + y*y + z*z); }
  cargo:warning=                                                                                                           ^
```

I bisected it down to merge commit 703ffa2f5d1d7b4d1dc60511be8badc0f52e0b26 ` Merge remote-tracking branch 'nixpkgs/staging-next' into staging`. Looks like it's related to conflict merge with rust version:
```diff
    Merge remote-tracking branch 'nixpkgs/staging-next' into staging

diff --cc pkgs/development/compilers/rust/1_56.nix
index b1d8167b9a0,00000000000..49df8bcd5ec
mode 100644,000000..100644
--- a/pkgs/development/compilers/rust/1_56.nix
+++ b/pkgs/development/compilers/rust/1_56.nix
@@@ -1,60 -1,0 +1,63 @@@
 +# New rust versions should first go to staging.
 +# Things to check after updating:
 +# 1. Rustc should produce rust binaries on x86_64-linux, aarch64-linux and x86_64-darwin:
 +#    i.e. nix-shell -p fd or @GrahamcOfBorg build fd on github
 +#    This testing can be also done by other volunteers as part of the pull
 +#    request review, in case platforms cannot be covered.
 +# 2. The LLVM version used for building should match with rust upstream.
 +#    Check the version number in the src/llvm-project git submodule in:
 +#    https://github.com/rust-lang/rust/blob/<version-tag>/.gitmodules
 +# 3. Firefox and Thunderbird should still build on x86_64-linux.
 +
 +{ stdenv, lib
 +, buildPackages
 +, newScope, callPackage
 +, CoreFoundation, Security, SystemConfiguration
 +, pkgsBuildTarget, pkgsBuildBuild, pkgsBuildHost
 +, makeRustPlatform
 +, llvmPackages_11
 +, llvmPackages_13, llvm_13
 +} @ args:
 +
 +import ./default.nix {
 +  rustcVersion = "1.56.1";
 +  rustcSha256 = "04cmqx7nn63hzz7z27b2b0dj2qx18rck9ifvip43s6dampx8v2f3";
 +
 +  llvmSharedForBuild = pkgsBuildBuild.llvmPackages_13.libllvm.override { enableSharedLibraries = true; };
 +  llvmSharedForHost = pkgsBuildHost.llvmPackages_13.libllvm.override { enableSharedLibraries = true; };
 +  llvmSharedForTarget = pkgsBuildTarget.llvmPackages_13.libllvm.override { enableSharedLibraries = true; };
 +
 +  llvmBootstrapForDarwin = llvmPackages_11;
 +
 +  # For use at runtime
 +  llvmShared = llvm_13.override { enableSharedLibraries = true; };
 +
++  # Expose llvmPackages used for rustc from rustc via passthru for LTO in Firefox
++  llvmPackagesForBuild = pkgsBuildBuild.llvmPackages_13;
++
 +  # Note: the version MUST be one version prior to the version we're
 +  # building
 +  bootstrapVersion = "1.55.0";
 +
 +  # fetch hashes by running `print-hashes.sh ${bootstrapVersion}`
 +  bootstrapHashes = {
 +    i686-unknown-linux-gnu = "6e42b6c44d2eb4170f4144423fa3c33338d8d5c3ea00b03bbac200c877bc9e98";
 +    x86_64-unknown-linux-gnu = "2080253a2ec36ac8ed6e060d30802d888533124b8d16545cfd4af898b365eaac";
 +    x86_64-unknown-linux-musl = "f24f68587253c4bfbe59d3d10fe4897068d9130538de6b2d02097a25718030c2";
 +    arm-unknown-linux-gnueabihf = "483444153d35cda51c6aec2c24bc4c97fa4fd30b28df4b60bf9763bd6e06da3a";
 +    armv7-unknown-linux-gnueabihf = "8c72f0eb75b10db970fb546c3b41f5e97df294d5dbbf0b8fa96e17f2b281ee9c";
 +    aarch64-unknown-linux-gnu = "eebdb2e659ed14884a49f0457d44e5e8c9f89fca3414533752c6dbb96232c156";
 +    aarch64-unknown-linux-musl = "2ce36a7d34f1f2aa43b4cbc0b437d96eefb45743828bf9ae699ff581ae257f28";
 +    x86_64-apple-darwin = "2e345ac7724c192c9487a2c6bd4f6c52c884d791981510288830d27d9a0bf2f3";
 +    aarch64-apple-darwin = "70c71d30d0de76912fcd88d503a6cb4323cfe6250c1a255be7e0d4e644b3d40a";
 +    powerpc64le-unknown-linux-gnu = "12bf6447d338cbe2b55539b84e6369b17e7eefe938d1ba7e3dd69781c9cc9812";
 +    riscv64gc-unknown-linux-gnu = "effceb45346fef3b0b54b357336e6f374f788b803bb1bee4084f25eace8907f3";
 +  };
 +
 +  selectRustPackage = pkgs: pkgs.rust_1_56;
 +
 +  rustcPatches = [
 +  ];
 +}
 +
 +(builtins.removeAttrs args [ "fetchpatch" "pkgsBuildHost" "llvmPackages_11" "llvmPackages_13" "llvm_13"])
diff --cc pkgs/top-level/all-packages.nix
index 043edd8b4f5,5bd902163ea..fea4a150de9
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@@ -12678,29 -12688,18 +12688,18 @@@ with pkgs
      inherit (darwin) apple_sdk;
    };

-   # Because rustc-1.46.0 enables static PIE by default for
-   # `x86_64-unknown-linux-musl` this release will suffer from:
-   #
-   # https://github.com/NixOS/nixpkgs/issues/94228
-   #
-   # So this commit doesn't remove the 1.45.2 release.
-   rust_1_45 = callPackage ../development/compilers/rust/1_45.nix {
-     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security SystemConfiguration;
-     llvm_10 = llvmPackages_10.libllvm;
-   };
 -  rust_1_55 = callPackage ../development/compilers/rust/1_55.nix {
 +  rust_1_56 = callPackage ../development/compilers/rust/1_56.nix {
      inherit (darwin.apple_sdk.frameworks) CoreFoundation Security SystemConfiguration;
 -    llvm_12 = llvmPackages_12.libllvm;
 +    llvm_13 = llvmPackages_13.libllvm;
    };
 -  rust = rust_1_55;
 +  rust = rust_1_56;

    mrustc = callPackage ../development/compilers/mrustc { };
    mrustc-minicargo = callPackage ../development/compilers/mrustc/minicargo.nix { };
    mrustc-bootstrap = callPackage ../development/compilers/mrustc/bootstrap.nix { };

-   rustPackages_1_45 = rust_1_45.packages.stable;
 -  rustPackages_1_55 = rust_1_55.packages.stable;
 -  rustPackages = rustPackages_1_55;
 +  rustPackages_1_56 = rust_1_56.packages.stable;
 +  rustPackages = rustPackages_1_56;

    inherit (rustPackages) cargo clippy rustc rustPlatform;
    
```

/cc @alyssais @yu-re-ka

**Comment by trofi** [comment]
Rolled back locally 6594da569b741ebce4598177050cbcf9f6aacd7c `rustc: 1.55.0 -> 1.56.0` (and fixed minor conflict). That made `firefox-94` building again for me.

**Comment by mweinelt** [comment]
I assume this is due to the move from llvm12 to llvm13, which firefox inherits from rustc.llvmPackages since a0201d653ed480fe076b7f89a4eb1fa136f14b8b.

**Comment by mweinelt** [comment]
Works when downgrading downgrading rust & firefox to llvm12. 

A comment in the firefox package says we should keep rustc.llvm aligned with the llvm version we built firefox with for LTO. 

https://github.com/NixOS/nixpkgs/blob/79ea7aa9f9cbb493e335d71a404d57488735df8d/pkgs/applications/networking/browsers/firefox/common.nix#L98

Would downgrading the llvm used for rust be acceptable for now and we revisit this issue after the next firefox release? @alyssais 

```
diff --git a/pkgs/development/compilers/rust/1_56.nix b/pkgs/development/compilers/rust/1_56.nix
index 49df8bcd5ec..c5c679131eb 100644
--- a/pkgs/development/compilers/rust/1_56.nix
+++ b/pkgs/development/compilers/rust/1_56.nix
@@ -16,24 +16,24 @@
 , pkgsBuildTarget, pkgsBuildBuild, pkgsBuildHost
 , makeRustPlatform
 , llvmPackages_11
-, llvmPackages_13, llvm_13
+, llvmPackages_12, llvm_12
 } @ args:
 
 import ./default.nix {
   rustcVersion = "1.56.1";
   rustcSha256 = "04cmqx7nn63hzz7z27b2b0dj2qx18rck9ifvip43s6dampx8v2f3";
 
-  llvmSharedForBuild = pkgsBuildBuild.llvmPackages_13.libllvm.override { enableSharedLibraries = true; };
-  llvmSharedForHost = pkgsBuildHost.llvmPackages_13.libllvm.override { enableSharedLibraries = true; };
-  llvmSharedForTarget = pkgsBuildTarget.llvmPackages_13.libllvm.override { enableSharedLibraries = true; };
+  llvmSharedForBuild = pkgsBuildBuild.llvmPackages_12.libllvm.override { enableSharedLibraries = true; };
+  llvmSharedForHost = pkgsBuildHost.llvmPackages_12.libllvm.override { enableSharedLibraries = true; };
+  llvmSharedForTarget = pkgsBuildTarget.llvmPackages_12.libllvm.override { enableSharedLibraries = true; };
 
   llvmBootstrapForDarwin = llvmPackages_11;
 
   # For use at runtime
-  llvmShared = llvm_13.override { enableSharedLibraries = true; };
+  llvmShared = llvm_12.override { enableSharedLibraries = true; };
 
   # Expose llvmPackages used for rustc from rustc via passthru for LTO in Firefox
-  llvmPackagesForBuild = pkgsBuildBuild.llvmPackages_13;
+  llvmPackagesForBuild = pkgsBuildBuild.llvmPackages_12;
 
   # Note: the version MUST be one version prior to the version we're
   # building
@@ -60,4 +60,4 @@ import ./default.nix {
   ];
 }
 
-(builtins.removeAttrs args [ "fetchpatch" "pkgsBuildHost" "llvmPackages_11" "llvmPackages_13" "llvm_13"])
+(builtins.removeAttrs args [ "fetchpatch" "pkgsBuildHost" "llvmPackages_11" "llvmPackages_12" "llvm_12"])
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 7d8c48e7532..9a228b51c6e 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -12686,7 +12686,7 @@ with pkgs;
 
   rust_1_56 = callPackage ../development/compilers/rust/1_56.nix {
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security SystemConfiguration;
-    llvm_13 = llvmPackages_13.libllvm;
+    llvm_12 = llvmPackages_12.libllvm;
   };
   rust = rust_1_56;
```

**Comment by alyssais** [comment]
I'd be very nervous about it — Rust tends to depend pretty tightly on a specific LLVM version, aiui.  I'd be more confident in things working if we just downgraded Rust, but the current release is in the unusual situation of having just had a high-profile security patch, which wasn't backported to any older versions.

Has this been reported upstream?  Is there maybe a Firefox patch they could use?  Presumably it won't just be us having this problem.

**Comment by mweinelt** [comment]
Reported just now over here:
https://bugzilla.mozilla.org/show_bug.cgi?id=1739913

**Comment by alyssais** [comment]
Great, thanks — let's see if we get anything there.

If not, and this becomes the last blocker for a staging merge or something, I think the best way forward would be to just re-introduce Rust 1.55 and use it only for building Firefox(/Thunderbird?) until FF can build with LLVM 13.

**Comment by msfjarvis** [comment]
Rust 1.56.0 also stabilises the 2021 edition which more and more crates are starting to depend on, and being on 1.55 blocks updates for those packages. One example of that is [starship](https://github.com/NixOS/nixpkgs/blob/05abd836d291cadd93d18fdb03898e26f1df7a77/pkgs/tools/misc/starship/default.nix), which uses the 2021 edition in [v1.0.0](https://github.com/starship/starship/releases/tag/v1.0.0).

**Comment by oxalica** [comment]
I think it's a packaging issue of llvm13, not rust.

I extract the llvm13-stdenv from firefox, and it cannot compile a simple C++ program. Changing to `llvmPackages_13` to 12 works fine.
```nix
{ pkgs ? import ./. {} }:
with pkgs;
let
  # llvmPkgs = llvmPackages_13.override { bootBintoolsNoLibc = null; bootBintools = null; }; # firefox use this, but actually has no effect.
  llvmPkgs = llvmPackages_13;
  buildStdenv = overrideCC stdenv llvmPkgs.clangUseLLVM;
in buildStdenv.mkDerivation {
  name = "test";
  dontUnpack = true;
  buildPhase = ''
    echo $CXX
    echo '#include <cmath>' >test.cc
    mkdir $out
    $CXX -c test.cc -o $out/test.o
  '';
  dontInstall = true;
}
```

<details>

<summary>Build log (not exactly the same but similar)</summary>

```
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
clang++
In file included from test.cc:1:
In file included from /nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/cmath:308:
In file included from /nix/store/l2kn6yyhxfz9rq8g0cl3za143wzr5s4k-libcxx-13.0.0-dev/include/c++/v1/math.h:300:
[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:38:12: [0m[0;1;31merror: [0m[1mno member named 'abs' in namespace 'std'[0m
using std::abs;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:39:12: [0m[0;1;31merror: [0m[1mno member named 'acos' in namespace 'std'[0m
using std::acos;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:40:12: [0m[0;1;31merror: [0m[1mno member named 'asin' in namespace 'std'[0m
using std::asin;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:41:12: [0m[0;1;31merror: [0m[1mno member named 'atan' in namespace 'std'[0m
using std::atan;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:42:12: [0m[0;1;31merror: [0m[1mno member named 'atan2' in namespace 'std'[0m
using std::atan2;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:43:12: [0m[0;1;31merror: [0m[1mno member named 'cos' in namespace 'std'[0m
using std::cos;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:44:12: [0m[0;1;31merror: [0m[1mno member named 'sin' in namespace 'std'[0m
using std::sin;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:45:12: [0m[0;1;31merror: [0m[1mno member named 'tan' in namespace 'std'[0m
using std::tan;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:46:12: [0m[0;1;31merror: [0m[1mno member named 'cosh' in namespace 'std'[0m
using std::cosh;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:47:12: [0m[0;1;31merror: [0m[1mno member named 'sinh' in namespace 'std'[0m
using std::sinh;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:48:12: [0m[0;1;31merror: [0m[1mno member named 'tanh' in namespace 'std'[0m
using std::tanh;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:49:12: [0m[0;1;31merror: [0m[1mno member named 'exp' in namespace 'std'[0m
using std::exp;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:50:12: [0m[0;1;31merror: [0m[1mno member named 'frexp' in namespace 'std'[0m
using std::frexp;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:51:12: [0m[0;1;31merror: [0m[1mno member named 'ldexp' in namespace 'std'[0m
using std::ldexp;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:52:12: [0m[0;1;31merror: [0m[1mno member named 'log' in namespace 'std'[0m
using std::log;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:53:12: [0m[0;1;31merror: [0m[1mno member named 'log10' in namespace 'std'[0m
using std::log10;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:54:12: [0m[0;1;31merror: [0m[1mno member named 'modf' in namespace 'std'[0m
using std::modf;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:55:12: [0m[0;1;31merror: [0m[1mno member named 'pow' in namespace 'std'[0m
using std::pow;
[0;1;32m      ~~~~~^
[0m[1m/nix/store/a87yinndp3fgpxxfxb9fyr9aln5yzsr7-gcc-10.3.0/lib64/gcc/x86_64-unknown-linux-gnu/10.3.0/../../../../include/c++/10.3.0/math.h:56:12: [0m[0;1;31merror: [0m[1mno member named 'sqrt' in namespace 'std'[0m
using std::sqrt;
[0;1;32m      ~~~~~^
[0m[0;1;31mfatal error: [0m[1mtoo many errors emitted, stopping now [-ferror-limit=][0m
20 errors generated.
```

</details>

cc: @lovek323 @7c6f434c  @dtzWill @primeos

**Comment by vcunat** [comment]
Let me close manually instead of waiting for merge to master.

**Comment by kvtb** [comment]
how did you solve finally?
Firefox 110 requires Rust 1.65+ which requires LLVM 13+, so there is no more option to downgrade LLVM to 12

**Comment by vcunat** [comment]
It's marked as solved by the PR above: #145459
